src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -88,8 +94,23 @@ class GitVersionPlugin implements Plugin<Project> {
             // first to preserve this behavior in cases where this call would fail but native "git" call does not.
             new DescribeCommand(git.getRepository()).call()
 
-            return GitCli.runGitCommand(project.rootDir, "describe", "--tags", "--always", "--first-parent",
-                    "--match=${prefix}*")
+            /*
+             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
+             * support versions of git < 1.8.4
+             */
+            int depth = 0
+            String revList = GitCli.runGitCommand(project.rootDir, "rev-list", "--first-parent", "HEAD")
+            for (String rev : revList.split(System.getProperty("line.separator"))) {
+                String exactTag = GitCli.runGitCommand(project.rootDir, "describe", "--tags", "--exact-match",
+                        "--match=${prefix}*", rev)
+                if (exactTag != "") {
+                    return formatDescription(exactTag, depth, rev)
```

Comment:
Would prefer to inline this because it's such a short function and the lambda is not used anywhere else.